### PR TITLE
Adds 2 features to KInspector modules

### PR DESCRIPTION
**Pull request by kenticoondrejp**
Extends PagesAnalyzerModule by adding column that tells whether there are any images without alt attribute on a page.
Extends Sitemap module by adding two columns for SEO metadata. Shows whether page has these metatags defined or inherits them from parent.

### CODE REVIEW

File: KInspector.Modules/Modules/General/PagesAnalyzerModule.cs [hunk 1]
```diff
@@ -383,6 +385,33 @@ private string GetHrefAttributeValue(string elementXml)
...
+                {
+                    if (altRegex.Match(match.Value).Success)
+                    {
+                        withAlt++;
+                    }
+                }
+                total++;
+            }
+
+            return string.Format("{0}/{1}", total-withAlt, total);
```
Comment (petrsvihlik) [inline KInspector.Modules/Modules/General/PagesAnalyzerModule.cs]: wouldn't it be better to replace the whole method with one line regex with a negative look-ahead? working examples here: 
http://stackoverflow.com/a/31887235/1332034
http://stackoverflow.com/a/14884088/1332034

File: KInspector.Modules/Modules/General/SiteMapModule.cs [hunk 2]
```diff
@@ -101,7 +103,8 @@ private object GetEffectiveColumnResult(DataTable table, DataRow row, string col
             do
             {
                 var rowValue = row[columnName];
-                if (rowValue.GetType() == columnType)
+                
+                if (rowValue.GetType() == columnType && ((columnName != "SEODescription" && columnName != "SEOKeyWords") || rowValue.ToString() != "EMPTY"))
```
Comment (petrsvihlik) [inline KInspector.Modules/Modules/General/SiteMapModule.cs]: It's not clear why have you added the conditions. Either add comment and describe the logic or, preferably, refactor the it - from the first glance it looks like a single-purpose piece code (and add comments as well :)).

**Comment by petrsvihlik** [comment]
Thanks @kenticoondrejp for submitting the PR.
Apart from the comments above, I have the following remarks:

 - Please always open an issue before you submit a PR. It helps us keep track of what's in the release and discuss things first (before they get implemented). Can you pls do it retrospectively for this issue?
 - Please always submit 1 PR per feature next time
 - while you are at it. Can you check why the `GetAppleTouchIconPrecomposedHrefs` method is not used?
 - when testing the PagesAnalyzerModule module I discovered a [bug](https://github.com/Kentico/KInspector/issues/106). Perhaps you'll be interested in looking into it in a separate PR...?

**Comment by kenticoondrejp** [comment]
This PR is ready for review.

File: KInspector.Modules/Scripts/PagesAnalyzerModule.sql [hunk 3]
```diff
@@ -1,4 +1,8 @@
 ﻿SELECT COALESCE(NULLIF(DocumentUrlPath, ''), NodeAliasPath) AS AliasPath, 
+	CASE
+			WHEN DocumentMenuRedirectToFirstChild = '1' THEN 'YES'
```
Comment (petrsvihlik) [inline KInspector.Modules/Scripts/PagesAnalyzerModule.sql]: please check for presence of `DocumentMenuRedirectToFirstChild ` column...it's definitely not there in v7. some server-side post processing will probably be required to preserve compatibility.

File: KInspector.Modules/Modules/General/PagesAnalyzerModule.cs [hunk 4]
```diff
@@ -388,28 +388,15 @@ private string GetHrefAttributeValue(string elementXml)
...
+            var regex = new Regex(regexPattern, RegexOptions.IgnoreCase);            
 
-            var imgRegex = new Regex(imgRegexPattern, RegexOptions.IgnoreCase | RegexOptions.Multiline);
-            var altRegex = new Regex(altRegexPattern, RegexOptions.IgnoreCase | RegexOptions.Multiline);
-
-            var withAlt = 0;
-            var total = 0;
-
-            foreach (Match match in imgRegex.Matches(html))
+            if (regex.IsMatch(html))
```
Comment (petrsvihlik) [inline KInspector.Modules/Modules/General/PagesAnalyzerModule.cs]: optional: you could even return the match count

File: KInspector.Modules/Modules/General/SiteMapModule.cs [hunk 5]
```diff
@@ -104,19 +113,39 @@ private object GetEffectiveColumnResult(DataTable table, DataRow row, string col
...
+                if (rowValue.ToString() != "EMPTY")
                 {
-                    // We have to search in parent node
-                    row = table.Select("NodeID = " + row["NodeParentID"]).FirstOrDefault();
+                    return rowValue;
                 }
+
+                row = table.Select("NodeID = " + row["NodeParentID"]).FirstOrDefault();
+
+            // as long as there is a parent
```
Comment (petrsvihlik) [inline KInspector.Modules/Modules/General/SiteMapModule.cs]: please start comments with a capital letter

**Comment by petrsvihlik** [comment]
the refactored code makes much more sense now!
i've added a few last inline comments and here are two extra:
+ please remove `var resultList = new List<string>();` - it's not used
+ `Type columnType` is not used in the `GetParentSeoDefinition` method (either use it or remove it)

**Comment by petrsvihlik** [comment]
Fixes #107 
Fixes #108

**Comment by kenticoondrejp** [comment]
Ready for review :)

**Comment by petrsvihlik** [comment]
TY, looks good. I secretly wished you would avoid using exec as it makes the sql less readable, but we can live with that...

**Pull request merged**